lib/firestarter/app_builder.rb

```diff
@@ -308,13 +308,19 @@ def setup_metrics
       copy_file "metrics.reek", ".metrics.reek"
     end
 
+    def override_gemfile_lock_for_tests
+      return unless ENV["TESTING"]
```

Comment:
I'm not a big fan of this, because we are writing "production" code only for the purpose of testing. The precedent was set by `override_path_for_tests`, but I just removed it in this branch, because there is no need for it to be in the `app_builder`.

Also, I'm concerned that because we have a `Gemfile.lock` we won't be testing in the same exact scenario it will be run, which kind of defeats the point of testing.

I understand that it is a bit painful to have to wait so long for tests, but I don't think this is the way to speed that up and still have a reliable test suite. We might be able to get around this concern by only using the `Gemfile.lock` locally, and not in CI, but I don't see how to get around the fact that this is code just for the tests.